src/hooks/useStatsig.tsx

```diff
@@ -1,17 +1,20 @@
+import { useEffect, useState } from 'react';
+
 import { StatsigClient } from '@statsig/js-client';
 import { StatsigProvider, useStatsigClient } from '@statsig/react-bindings';
 
-export enum StatSigFlags {
-  // When adding a flag here, make sure to add an analytics tracker in useAnalytics.ts
-  ffSkipMigration = 'ff_skip_migration',
-}
+import { StatSigFlags, statsigClientPromise } from '@/lib/statsig';
 
 export const StatSigProvider = ({ children }: { children: React.ReactNode }) => {
-  const client = new StatsigClient(`${import.meta.env.VITE_STATSIG_CLIENT_KEY}`, {
-    // TODO: fill in with ip address
-  });
-  client.initializeSync();
-
+  const [client, setClient] = useState<StatsigClient | null>(null);
+  useEffect(() => {
+    const setAsyncClient = async () => {
+      const statsigClient = await statsigClientPromise;
+      setClient(statsigClient);
+    };
+    setAsyncClient();
+  }, [statsigClientPromise]);
+  if (!client) return null;
```

Comment:
we don't want the app to render until statsig is done loading, otherwise it will break stuff - because abacus lib depends on statsig.